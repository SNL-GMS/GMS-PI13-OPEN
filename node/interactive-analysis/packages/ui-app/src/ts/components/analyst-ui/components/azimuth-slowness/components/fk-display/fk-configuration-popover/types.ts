import { FkTypes } from '@gms/common-graphql';
import { CheckboxSearchListTypes } from '@gms/ui-core-components';
import { FkConfigurationWithUnits, FkUnits } from '~analyst-ui/components/azimuth-slowness/types';

export enum FkConfigurationPopoverPanel {
  DEFAULT,
  ADVANCED
}

export interface FkConfigurationPopoverState {
  openPanel: FkConfigurationPopoverPanel;
  fkUnits: FkUnits;
  normalizeWaveforms: boolean;
  mediumVelocity: number;
  maximumSlowness: number;
  numberOfPoints: number;
  channelCheckboxes: CheckboxSearchListTypes.CheckboxItem[];
  useVerticalChannelOffsets: boolean;
}
export interface FkConfigurationPopoverProps extends FkTypes.FkConfiguration {
  fkUnitDisplayed: FkUnits;
  applyFkConfiguration(configuration: FkConfigurationWithUnits);
  close();
}
