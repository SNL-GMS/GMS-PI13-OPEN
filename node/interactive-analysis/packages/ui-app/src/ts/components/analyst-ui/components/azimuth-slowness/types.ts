import {
  CommonTypes,
  ConfigurationTypes,
  EventTypes,
  FkTypes,
  ProcessingStationTypes,
  SignalDetectionTypes,
  WaveformTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { Client } from '@gms/ui-apollo';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { ChildProps, MutationFunction } from 'react-apollo';
import { FilterType, FkThumbnailSize } from './components/fk-thumbnail-list/fk-thumbnails-controls';
import { LeadLagPairs } from './constants';

export interface FkParams {
  windowParams: FkTypes.WindowParameters;
  frequencyPair: FkTypes.FrequencyBand;
}

export interface LeadLagPairAndString {
  leadLagPairs: LeadLagPairs;
  windowParams: FkTypes.WindowParameters;
}

export enum FkUnits {
  FSTAT = 'FSTAT',
  POWER = 'POWER'
}

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export interface FkConfigurationWithUnits extends FkTypes.FkConfiguration {
  fkUnitToDisplay: FkUnits;
}

/**
 * Azimuth Slowness Redux Props
 */
export interface AzimuthSlownessReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  client: Client;
  currentStageInterval: AnalystWorkspaceTypes.StageInterval;
  currentTimeInterval: CommonTypes.TimeRange;
  selectedSdIds: string[];
  openEventId: string;
  sdIdsToShowFk: string[];
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  setSelectedSdIds(ids: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * Azimuth Slowness State
 */
export interface AzimuthSlownessState {
  fkThumbnailSizePx: FkThumbnailSize;
  fkThumbnailColumnSizePx: number;
  filterType: FilterType;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  userInputFkFrequency: FkTypes.FrequencyBand;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  fkInnerContainerWidthPx: number;
  fkFrequencyThumbnails: Immutable.Map<string, FkTypes.FkFrequencyThumbnail[]>;
}

/**
 * Mutations used by the Az Slow display
 */
export interface AzimuthSlownessMutations {
  computeFks: MutationFunction<{}>;
  computeFkFrequencyThumbnails: MutationFunction<{}>;
  setWindowLead: MutationFunction<{}>;
  markFksReviewed: MutationFunction<{}>;
}

/**
 * Consolidated props for Azimuth Slowness
 */
export type AzimuthSlownessProps = AzimuthSlownessReduxProps &
  ChildProps<AzimuthSlownessMutations> &
  ProcessingStationTypes.DefaultStationsQueryProps &
  ConfigurationTypes.UIConfigurationQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  EventTypes.EventsInTimeRangeQueryProps;

/**
 * State of the az slow panel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
}

export interface AzimuthSlownessPanelProps {
  // Data
  defaultStations: ProcessingStationTypes.ProcessingStation[];
  eventsInTimeRange: EventTypes.Event[];
  displayedSignalDetection: SignalDetectionTypes.SignalDetection | undefined;
  openEvent: EventTypes.Event | undefined;
  associatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  signalDetectionsToDraw: SignalDetectionTypes.SignalDetection[];
  signalDetectionsIdToFeaturePredictions: Immutable.Map<string, EventTypes.FeaturePrediction[]>;
  signalDetectionsByStation: SignalDetectionTypes.SignalDetection[];
  featurePredictionsForDisplayedSignalDetection: EventTypes.FeaturePrediction[];
  distances: EventTypes.LocationToStationDistance[];
  // Redux state as props
  selectedSdIds: string[];
  defaultWaveformFilters: WaveformTypes.WaveformFilter[];
  sdIdsToShowFk: string[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  fkFrequencyThumbnails: FkTypes.FkFrequencyThumbnail[];
  // Azimuth display state as props
  fkThumbnailColumnSizePx: number;
  fkDisplayWidthPx: number;
  fkDisplayHeightPx: number;
  filterType: FilterType;
  fkThumbnailSizePx: FkThumbnailSize;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  numberOfOutstandingComputeFkMutations: number;
  userInputFkFrequency: FkTypes.FrequencyBand;
  fkUnitForDisplayedSignalDetection: FkUnits;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  fkInnerContainerWidthPx: number;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  // Prop functions
  adjustFkInnerContainerWidth(
    fkThumbnailsContainer: HTMLDivElement,
    fkThumbnailsInnerContainer: HTMLDivElement
  ): void;
  markFksForSdIdsAsReviewed(sdIds: string[]): void;
  updateFkThumbnailSize(size: FkThumbnailSize): void;
  updateFkFilter(filterType: FilterType): void;
  setFkThumbnailColumnSizePx(newSizePx: number): void;
  computeFkAndUpdateState(fkInput: FkTypes.FkInput): void;
  changeUserInputFks(
    windowParams: FkTypes.WindowParameters,
    frequencyBand: FkTypes.FrequencyBand
  ): void;
  setFkUnitForSdId(sdId: string, fkUnit: FkUnits): void;

  // Redux functions
  setSelectedSdIds(sdIds: string[]): void;
  setSdIdsToShowFk(sdIds: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}
