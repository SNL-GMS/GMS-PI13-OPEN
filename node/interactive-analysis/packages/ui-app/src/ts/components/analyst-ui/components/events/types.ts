import {
  CommonTypes,
  EventTypes,
  ProcessingStationTypes,
  SignalDetectionTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import { ChildProps, MutationFunction } from 'react-apollo';
import { gmsColors, semanticColors } from '~scss-config/color-preferences';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}
export const eventFilterToColorMap: Map<any, string> = new Map();
eventFilterToColorMap.set(EventFilters.EDGE, gmsColors.gmsBackground);
eventFilterToColorMap.set(EventFilters.COMPLETED, semanticColors.analystComplete);

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction<{}>;
  saveEvent: MutationFunction<{}>;
  saveAllModifiedEvents: MutationFunction<{}>;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}
export interface SignalDetectionHypothesisWithStation
  extends SignalDetectionTypes.SignalDetectionHypothesis {
  stationName?: string;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  EventTypes.EventsInTimeRangeQueryProps &
  ProcessingStationTypes.DefaultStationsQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  CommonTypes.WorkspaceStateProps;
