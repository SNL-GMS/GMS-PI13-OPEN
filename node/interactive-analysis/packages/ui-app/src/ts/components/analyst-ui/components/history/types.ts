import { CacheTypes, CommonTypes, EventTypes } from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import React from 'react';
import { ChildProps, MutationFunction } from 'react-apollo';
import { HistoryEntryAction } from './history-stack/types';

/**
 * Mutations used by the History component
 */
export interface HistoryMutations {
  undoHistory: MutationFunction<CacheTypes.UndoHistoryMutationResult>;
  redoHistory: MutationFunction<CacheTypes.RedoHistoryMutationResult>;
  undoHistoryById: MutationFunction<CacheTypes.UndoHistoryByIdMutationResult>;
  redoHistoryById: MutationFunction<CacheTypes.RedoHistoryByIdMutationResult>;
  undoEventHistory: MutationFunction<CacheTypes.UndoEventHistoryMutationResult>;
  redoEventHistory: MutationFunction<CacheTypes.RedoEventHistoryMutationResult>;
  undoEventHistoryById: MutationFunction<CacheTypes.UndoEventHistoryByIdMutationResult>;
  redoEventHistoryById: MutationFunction<CacheTypes.RedoEventHistoryByIdMutationResult>;
}

/**
 * History Redux Props
 */
export interface HistoryReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  historyActionInProgress: number;
  openEventId: string;
  currentTimeInterval: CommonTypes.TimeRange;
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  keyPressActionQueue: AnalystWorkspaceTypes.KeyAction[];
  setKeyPressActionQueue(actions: AnalystWorkspaceTypes.KeyAction[]): void;
  incrementHistoryActionInProgress();
  decrementHistoryActionInProgress();
}

/**
 * History Props
 */
export type HistoryProps = HistoryReduxProps &
  ChildProps<HistoryMutations> &
  CacheTypes.HistoryQueryProps &
  EventTypes.EventsInTimeRangeQueryProps;

/**
 * State for the HistoryComponent
 */
export interface HistoryState {
  actionIntentPointer: HistoryEntryPointer;
}
/**
 * History Panel Props
 * optional non-ideal-state element to render in the children history-stacks
 */
export interface HistoryPanelProps {
  nonIdealState?: NonIdealHistoryState;
}

/**
 * Keeps track of which HistoryEntryAction is being target for an undo/redo action
 * and provides a check function isIncluded for checking if other actions would be
 * included based on whether the target is an event or global undo/redo.
 */
export interface HistoryEntryPointer {
  entryId: string;
  entryType: HistoryEntryAction;
  isEventMode: boolean;
  isIncluded?(entry: CacheTypes.History): boolean;
  isChangeIncluded(change: CacheTypes.HistoryChange): boolean;
}

/**
 * The type for the data used by history context
 */
export interface HistoryContextData {
  glContainer: GoldenLayout.Container;
  historyList: CacheTypes.History[];
  historyActionIntent: HistoryEntryPointer;
  openEventId: string;
  eventsInTimeRange: EventTypes.Event[];
  historyActionInProgress: number;
  redoEventHistory(quantity?: number): void;
  redoEventHistoryById(id: string): void;
  redoHistory(quantity?: number): void;
  redoHistoryById(id: string): void;
  setHistoryActionIntent(pointer: HistoryEntryPointer): void;
  undoEventHistory(quantity?: number): void;
  undoEventHistoryById(id: string): void;
  undoHistory(quantity?: number): void;
  undoHistoryById(id: string): void;
}

/**
 * History Context to provide access to these high-level props to the History-Component tree.
 * Here, we instantiate the Context variable and set up the defaults for the history context
 */
export const HistoryContext: React.Context<HistoryContextData> = React.createContext<
  HistoryContextData
>({
  glContainer: undefined,
  openEventId: undefined,
  eventsInTimeRange: [],
  undoEventHistory: undefined,
  undoEventHistoryById: undefined,
  redoEventHistory: undefined,
  redoEventHistoryById: undefined,
  undoHistory: undefined,
  undoHistoryById: undefined,
  redoHistory: undefined,
  redoHistoryById: undefined,
  historyActionIntent: undefined,
  setHistoryActionIntent: undefined,
  historyActionInProgress: 0,
  historyList: []
});

export interface NonIdealHistoryState {
  loadingEvent: JSX.Element;
  loadingHistory: JSX.Element;
}
