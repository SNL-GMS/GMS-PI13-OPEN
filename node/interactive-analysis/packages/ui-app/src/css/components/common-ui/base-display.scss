.base-display {
  display: flex;
  position: relative;
  flex-direction: column;
  &.full-width-height {
    height: 100%;
    width: 100%;
    flex-grow: 1;
  }
}
