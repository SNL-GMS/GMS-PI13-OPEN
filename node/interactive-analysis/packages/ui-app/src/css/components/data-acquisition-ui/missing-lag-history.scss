.lag-history-display,
.missing-history-display {
  .soh-legend {
    // # is passing through the contents as a string
    height: calc(100% - #{$toolbar-height} - 1rem);
    top: initial !important;
  }
  .legend-and-charts {
    @include flexColumn;
    flex-grow: 1;
    height: auto;
  }
  .core-chart {
    flex-grow: 1;
    overflow-y: auto;
    height: auto;
    max-width: 100%;
  }
}
