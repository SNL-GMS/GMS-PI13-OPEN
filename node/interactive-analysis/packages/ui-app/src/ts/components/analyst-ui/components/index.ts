export { ReduxApolloWaveformDisplay as WaveformDisplay } from './waveform-display';
export { Events } from './events';
export { History } from './history';
export { SignalDetections } from './signal-detections';
export { Workflow } from './workflow';
export { ReduxApolloMap as Map } from './map';
export { Location } from './location';
export { Magnitude } from './magnitude';

export { ReduxApolloAzimuthSlowness as AzimuthSlowness } from './azimuth-slowness/azimuth-slowness-container';
