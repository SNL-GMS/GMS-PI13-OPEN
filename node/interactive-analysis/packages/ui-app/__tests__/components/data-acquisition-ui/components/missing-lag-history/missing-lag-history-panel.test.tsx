import { SohTypes } from '@gms/common-graphql';
import { SohMonitorType } from '@gms/common-graphql/lib/graphql/soh/types';
import { ValueType } from '@gms/common-util/lib/types/value-type';
import { Client } from '@gms/ui-apollo';
import DefaultClient from 'apollo-boost';
import React from 'react';
import { ApolloProvider } from 'react-apollo';
import { act } from 'react-dom/test-utils';
import { BaseDisplayContext } from '../../../../../src/ts/components/common-ui/components/base-display';
import { MissingLagHistoryPanel } from '../../../../../src/ts/components/data-acquisition-ui/components/missing-lag-history';
import * as validateNonIdealStateDependency from '../../../../../src/ts/components/data-acquisition-ui/components/missing-lag-history/non-ideal-states';
import { MissingLagHistoryPanelProps } from '../../../../../src/ts/components/data-acquisition-ui/components/missing-lag-history/types';
import * as Util from '../../../../../src/ts/components/data-acquisition-ui/components/missing-lag-history/utils';
import { BarLineChartData } from '../../../../../src/ts/components/data-acquisition-ui/shared/chart/types';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');
const HUNDRED_MS = 2000;
const MOCK_TIME_TWO = 1530518207007;
// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();
// Setup
global.Date.now = jest.fn(() => MOCK_TIME_TWO);

const client: Client = new DefaultClient<any>();
const waitForComponentToPaint = async (wrapper: any) => {
  // fixes React warning that "An update to Component inside a test was not wrapped in act(...)."
  // this has something to do with use state or apollo and needs 100ms to figure itself out
  // tslint:disable-next-line: await-promise
  await act(async () => {
    await new Promise(resolve => setTimeout(resolve, HUNDRED_MS));
    wrapper.update();
  });
};
describe('Soh Lag panel', () => {
  const missingLagHistoryPanelProps: MissingLagHistoryPanelProps = {
    monitorType: SohTypes.SohMonitorType.MISSING,
    valueType: ValueType.PERCENTAGE,
    // tslint:disable-next-line: no-magic-numbers
    sohHistoricalDurations: [30000, 90000],
    station: {
      id: '1',
      uuid: '1',
      needsAcknowledgement: true,
      needsAttention: true,
      sohStatusSummary: undefined,
      stationGroups: [],
      statusContributors: [],
      time: undefined,
      stationName: '1',
      allStationAggregates: [],
      channelSohs: [
        {
          channelName: 'adsf',
          channelSohStatus: undefined,
          allSohMonitorValueAndStatuses: [
            {
              monitorType: SohMonitorType.LAG,
              value: 10,
              valuePresent: true,
              status: SohTypes.SohStatusSummary.GOOD,
              hasUnacknowledgedChanges: true,
              contributing: false,
              quietUntilMs: 1,
              thresholdBad: 3,
              thresholdMarginal: 3
            },
            {
              monitorType: SohMonitorType.LAG,
              value: 11,
              valuePresent: true,
              status: SohTypes.SohStatusSummary.GOOD,
              hasUnacknowledgedChanges: true,
              contributing: false,
              quietUntilMs: 1,
              thresholdBad: 3,
              thresholdMarginal: 3
            }
          ]
        },
        {
          channelName: 'adsf2',
          channelSohStatus: undefined,
          allSohMonitorValueAndStatuses: [
            {
              monitorType: SohMonitorType.LAG,
              value: 10,
              valuePresent: true,
              status: SohTypes.SohStatusSummary.GOOD,
              hasUnacknowledgedChanges: true,
              contributing: false,
              quietUntilMs: 1,
              thresholdBad: 3,
              thresholdMarginal: 3
            },
            {
              monitorType: SohMonitorType.LAG,
              value: 11,
              valuePresent: true,
              status: SohTypes.SohStatusSummary.GOOD,
              hasUnacknowledgedChanges: true,
              contributing: false,
              quietUntilMs: 1,
              thresholdBad: 3,
              thresholdMarginal: 3
            }
          ]
        }
      ]
    },
    sohStatus: {
      lastUpdated: 0,
      loading: false,
      error: undefined,
      stationAndStationGroupSoh: {
        isUpdateResponse: false,
        stationGroups: [],
        stationSoh: [
          {
            id: '1',
            uuid: '1',
            needsAcknowledgement: true,
            needsAttention: true,
            sohStatusSummary: undefined,
            stationGroups: [],
            statusContributors: [],
            time: undefined,
            stationName: '1',
            allStationAggregates: [],
            channelSohs: [
              {
                channelName: 'adsf',
                channelSohStatus: undefined,
                allSohMonitorValueAndStatuses: [
                  {
                    monitorType: SohMonitorType.LAG,
                    value: 10,
                    valuePresent: true,
                    status: SohTypes.SohStatusSummary.GOOD,
                    hasUnacknowledgedChanges: true,
                    contributing: false,
                    quietUntilMs: 1,
                    thresholdBad: 3,
                    thresholdMarginal: 3
                  },
                  {
                    monitorType: SohMonitorType.LAG,
                    value: 11,
                    valuePresent: true,
                    status: SohTypes.SohStatusSummary.GOOD,
                    hasUnacknowledgedChanges: true,
                    contributing: false,
                    quietUntilMs: 1,
                    thresholdBad: 3,
                    thresholdMarginal: 3
                  }
                ]
              }
            ]
          }
        ]
      }
    }
  };
  // mock the functions that create data that is passed to the charts
  jest
    .spyOn(validateNonIdealStateDependency, 'validateNonIdealState')
    .mockImplementation(() => undefined);
  const barLineChartData: BarLineChartData = {
    categories: { x: ['x1', 'x2'], y: [] },
    lineDefs: [
      {
        color: 'red',
        id: 1,
        value: [
          { x: 1, y: 1 },
          { x: 2, y: 2 }
        ]
      }
    ],
    barDefs: [
      { color: 'red', id: 3, value: { x: 1, y: 1 } },
      { color: 'red', id: 4, value: { x: 2, y: 2 } }
    ],
    thresholdsMarginal: [1, 2],
    thresholdsBad: [1, 2]
  };
  jest.spyOn(Util, 'getChartData').mockImplementation(() => barLineChartData);
  it('should be defined', () => {
    expect(MissingLagHistoryPanel).toBeDefined();
  });
  it('should match snapshot', async () => {
    const missingLagHistoryPanel = Enzyme.shallow(
      <ApolloProvider client={client}>
        <BaseDisplayContext.Provider
          value={{
            glContainer: { width: 100, height: 100 } as any,
            widthPx: 100,
            heightPx: 100
          }}
        >
          <MissingLagHistoryPanel {...missingLagHistoryPanelProps} />
        </BaseDisplayContext.Provider>
      </ApolloProvider>
    );
    // we gotta wait for the use state
    await waitForComponentToPaint(missingLagHistoryPanel);
    missingLagHistoryPanel.update();
    expect(missingLagHistoryPanel.dive()).toMatchSnapshot();
  });
});
