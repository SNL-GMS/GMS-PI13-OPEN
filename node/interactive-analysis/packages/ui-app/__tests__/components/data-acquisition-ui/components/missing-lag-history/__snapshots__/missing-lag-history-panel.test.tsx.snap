// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Soh Lag panel should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "client": DefaultClient {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "queryManager": QueryManager {
          "assumeImmutableResults": false,
          "clientAwareness": Object {
            "name": undefined,
            "version": undefined,
          },
          "dataStore": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "fetchQueryRejectFns": Map {},
          "idCounter": 1,
          "inFlightLinkObservables": Map {},
          "link": ApolloLink {
            "request": [Function],
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutationStore": MutationStore {
            "store": Object {},
          },
          "onBroadcast": [Function],
          "pollingInfoByQueryId": Map {},
          "queries": Map {},
          "queryDeduplication": true,
          "queryStore": QueryStore {
            "store": Object {},
          },
          "ssrMode": false,
          "transformCache": WeakMap {},
        },
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.6.4",
        "watchQuery": [Function],
      },
    }
  }
>
  <ContextProvider
    value={
      Object {
        "glContainer": Object {
          "height": 100,
          "width": 100,
        },
        "heightPx": 100,
        "widthPx": 100,
      }
    }
  >
    <Component
      monitorType="MISSING"
      sohHistoricalDurations={
        Array [
          30000,
          90000,
        ]
      }
      sohStatus={
        Object {
          "error": undefined,
          "lastUpdated": 0,
          "loading": false,
          "stationAndStationGroupSoh": Object {
            "isUpdateResponse": false,
            "stationGroups": Array [],
            "stationSoh": Array [
              Object {
                "allStationAggregates": Array [],
                "channelSohs": Array [
                  Object {
                    "allSohMonitorValueAndStatuses": Array [
                      Object {
                        "contributing": false,
                        "hasUnacknowledgedChanges": true,
                        "monitorType": "LAG",
                        "quietUntilMs": 1,
                        "status": "GOOD",
                        "thresholdBad": 3,
                        "thresholdMarginal": 3,
                        "value": 10,
                        "valuePresent": true,
                      },
                      Object {
                        "contributing": false,
                        "hasUnacknowledgedChanges": true,
                        "monitorType": "LAG",
                        "quietUntilMs": 1,
                        "status": "GOOD",
                        "thresholdBad": 3,
                        "thresholdMarginal": 3,
                        "value": 11,
                        "valuePresent": true,
                      },
                    ],
                    "channelName": "adsf",
                    "channelSohStatus": undefined,
                  },
                ],
                "id": "1",
                "needsAcknowledgement": true,
                "needsAttention": true,
                "sohStatusSummary": undefined,
                "stationGroups": Array [],
                "stationName": "1",
                "statusContributors": Array [],
                "time": undefined,
                "uuid": "1",
              },
            ],
          },
        }
      }
      station={
        Object {
          "allStationAggregates": Array [],
          "channelSohs": Array [
            Object {
              "allSohMonitorValueAndStatuses": Array [
                Object {
                  "contributing": false,
                  "hasUnacknowledgedChanges": true,
                  "monitorType": "LAG",
                  "quietUntilMs": 1,
                  "status": "GOOD",
                  "thresholdBad": 3,
                  "thresholdMarginal": 3,
                  "value": 10,
                  "valuePresent": true,
                },
                Object {
                  "contributing": false,
                  "hasUnacknowledgedChanges": true,
                  "monitorType": "LAG",
                  "quietUntilMs": 1,
                  "status": "GOOD",
                  "thresholdBad": 3,
                  "thresholdMarginal": 3,
                  "value": 11,
                  "valuePresent": true,
                },
              ],
              "channelName": "adsf",
              "channelSohStatus": undefined,
            },
            Object {
              "allSohMonitorValueAndStatuses": Array [
                Object {
                  "contributing": false,
                  "hasUnacknowledgedChanges": true,
                  "monitorType": "LAG",
                  "quietUntilMs": 1,
                  "status": "GOOD",
                  "thresholdBad": 3,
                  "thresholdMarginal": 3,
                  "value": 10,
                  "valuePresent": true,
                },
                Object {
                  "contributing": false,
                  "hasUnacknowledgedChanges": true,
                  "monitorType": "LAG",
                  "quietUntilMs": 1,
                  "status": "GOOD",
                  "thresholdBad": 3,
                  "thresholdMarginal": 3,
                  "value": 11,
                  "valuePresent": true,
                },
              ],
              "channelName": "adsf2",
              "channelSohStatus": undefined,
            },
          ],
          "id": "1",
          "needsAcknowledgement": true,
          "needsAttention": true,
          "sohStatusSummary": undefined,
          "stationGroups": Array [],
          "stationName": "1",
          "statusContributors": Array [],
          "time": undefined,
          "uuid": "1",
        }
      }
      valueType="PERCENTAGE"
    />
  </ContextProvider>
</ContextProvider>
`;
